mergify_engine/tests/functional/actions/test_queue.py

```diff
@@ -2585,6 +2587,974 @@ async def test_queue_manual_merge(self):
+                    "name": "low-priority",
+                    "reason": "test freeze reason",
+                    "user_app_name": "testing application",
+                }
+            ],
+            "description": 'The queue "low-priority" has been successfully frozen.',
+        }
+
+        r = await self.app.get(
+            f"/v1/repos/{config.TESTING_ORGANIZATION_NAME}/{self.RECORD_CONFIG['repository_name']}/queues/freezes",
+            headers={
+                "Authorization": f"bearer {self.api_key_admin}",
+                "Content-type": "application/json",
+            },
+        )
+
+        assert r.status_code == 200
+        json_data = r.json()
+        list_queue_name = [qf["name"] for qf in json_data["queue_freeze"]]
+        assert "default" in list_queue_name
+        assert "low-priority" in list_queue_name
+        assert json_data["description"] == "there is 2 queue(s) currently frozen."
+
+        queue_name = "default"
+
+        r = await self.app.delete(
+            f"/v1/repos/{config.TESTING_ORGANIZATION_NAME}/{self.RECORD_CONFIG['repository_name']}/queue/{queue_name}/unfreeze",
+            headers={
+                "Authorization": f"bearer {self.api_key_admin}",
+                "Content-type": "application/json",
+            },
+        )
+
+        assert r.status_code == 204
+
+        r = await self.app.get(
+            f"/v1/repos/{config.TESTING_ORGANIZATION_NAME}/{self.RECORD_CONFIG['repository_name']}/queues/freezes",
+            headers={
+                "Authorization": f"bearer {self.api_key_admin}",
+                "Content-type": "application/json",
+            },
+        )
+
+        assert r.status_code == 200
+        assert r.json() == {
+            "queue_freeze": [
+                {
+                    "name": "low-priority",
+                    "reason": "test freeze reason",
+                    "user_app_name": "testing application",
+                    "freeze_date": mock.ANY,
+                },
+            ],
+            "description": "there is 1 queue(s) currently frozen.",
+        }
+
+        queue_name = "low-priority"
+
+        r = await self.app.delete(
+            f"/v1/repos/{config.TESTING_ORGANIZATION_NAME}/{self.RECORD_CONFIG['repository_name']}/queue/{queue_name}/unfreeze",
+            headers={
+                "Authorization": f"bearer {self.api_key_admin}",
+                "Content-type": "application/json",
+            },
+        )
+
+        assert r.status_code == 204
+
+        r = await self.app.get(
+            f"/v1/repos/{config.TESTING_ORGANIZATION_NAME}/{self.RECORD_CONFIG['repository_name']}/queues/freezes",
+            headers={
+                "Authorization": f"bearer {self.api_key_admin}",
+                "Content-type": "application/json",
+            },
+        )
+
+        assert r.status_code == 404
+        assert r.json() == {
+            "detail": "There is no frozen queues to show.",
+        }
+
```

Comment:
None of the tests in this file actually test that the freeze *works*, e.g., that it does what it means.
We need a test that checks the PR is not merged and that the given reason makes sense for the user in what we report to GitHub.
And than then when we unfreeze, it merges again.